src/dev/build/tasks/patch_native_modules_task.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { scanCopy, untar, deleteAll } from '../lib';
+import { writeSync, openSync, existsSync, unlinkSync } from 'fs';
+import binaryInfo from 'nodegit/lib/utils/binary_info';
+import wreck from 'wreck';
+import mkdirp from 'mkdirp';
+import { dirname, join, basename } from 'path';
+
+async function download(url, destination, log) {
+  const response = await wreck.request('GET', url);
+
+  if (response.statusCode !== 200) {
+    throw new Error(
+      `Unexpected status code ${response.statusCode} when downloading ${url}`
+    );
+  }
+  mkdirp.sync(dirname(destination));
+  const fileHandle = openSync(destination, 'w');
+
+  await new Promise((resolve, reject) => {
+    response.on('data', chunk => {
+      writeSync(fileHandle, chunk);
+    });
+
+    response.on('error', (err) => {
+      if (existsSync(destination)) {
+        // remove the unfinished file
+        unlinkSync(destination);
+      }
+      reject(err);
+    });
+    response.on('end', () => {
+      log.debug('Downloaded ', url);
+      resolve();
+    });
+  });
+
+}
+
+async function downloadAndExtractTarball(url, dest, log, retry) {
+  try {
+    await download(url, dest, log);
+    const extractDir = join(dirname(dest), basename(dest, '.tar.gz'));
+    await untar(dest, extractDir, {
+      strip: 1
+    });
+    return extractDir;
+  } catch (e) {
+    if (retry > 0) {
+      await downloadAndExtractTarball(url, dest, log, retry - 1);
+    } else {
+      throw e;
+    }
+  }
+}
+
+async function patchNodeGit(config, log, build, platform) {
+  const plat = platform.isWindows() ? 'win32' : platform.getName();
+  const arch = platform.getNodeArch().split('-')[1];
+  const info = binaryInfo(plat, arch);
+  const downloadUrl = info.hosted_tarball;
+  const packageName = info.package_name;
+  const downloadPath = config.resolveFromRepo('.nodegit_binaries', packageName);
```

Comment:
done.

Reply:
@spalger Spencer, under what circumstances the module is located in the folder "x-pack / node_modules" instead of the folder "kibana / node_modules". I am worried that one day nodegit will be installed into x-pack/node_modules, so the current replacement path is wrong.